### ISSUE TITLE
serve static files from CDN

### ISSUE THREAD

**Issue by hnykda** [epidemics/covid#41]
Only if it's actually needed. We could reduce some load on the server if we hosted the static files elsewhere (JS and CSS). That would reduce responsivness of the site (userfriendlines) and reduce cost on compute cost and egress between the client and the server.

https://epidemicforecasting.slack.com/archives/C0100EK7N8N/p1584518497196200?thread_ts=1584472523.152600&cid=C0100EK7N8N

Solution:
* add `gsutil cp -Z -a <somecacheheaders as in epifor> ...` step which would take the content of `src/server/static` and upload it to the bucket (for staging and production independently), ideally two way sync
* setup the `static_url` to be served from GCS in deployments, maintain the file serving locally, that likely means dealing with `url_for` , maybe https://fastapi.tiangolo.com/tutorial/static-files/  and changing url_for from `/static` to something like `https://storage.googleapis.com/static-covid/static/`

**Comment by gavento** [comment]
Note: Main data files (JSON) are already served from GCS storage, which scales extremely well. We can upload assets too, I guess?

**Comment by jez4** [comment]
@hnykda I am checking the deploy action in pythonapp.yml. I should add the gsutil cp there?

**Comment by hnykda** [comment]
Yes, it should be there on a deploy step. The command is IMO equivalent to the `push_to_bucket

The hard part seems to be actually rendering templates with different static URL paths for static files (I asked about it here https://github.com/encode/starlette/issues/884).

In the first case, I propose to use something like:

```
APP_DEBUG = True  # local dev
STATIC_URL= "/static" if APP_DEBUG else "https://storage.googleapis.com/static-covid/static/"

...
def someview():
    return Template("<template>.html", {static_prefix=STATIC_URL})
```

**Comment by hnykda** [comment]
@jez4 I just realized that I did something in the past (it's not in master anymore): https://github.com/epidemics/covid/pull/122/files

^ you IMO want to take this and move it to the deploy branch only (so it's iterated only on deploy) and instead of setting that public access and cache control actually uploading everything gzupped.

**Comment by jez4** [comment]
I should use {{ STATIC_URL }} for templates

**Comment by jez4** [comment]
I have managed to upload static files to the bucket gzipped. Now the templates.

**Comment by jez4** [comment]
The templates are done. Now just to perfect the workflow. I will ask for review afterwards.

**Comment by hnykda** [comment]
Closed via #333
